### DISCUSSION TITLE
Image links generated by quarto render won't  show up in web pages

### DISCUSSION THREAD

**Discussion by prvst** [quarto-dev/quarto-cli#3921] [Q&A]
Hi, I'm adapting my hugo/blogdown website to use Quarto `qmd` files. I'm following the instructions rendering my `.qmd` file into an `.md`, and the website is working fine with one exception; the output of ggplot commands is not showing up in the webpages.

Looking at my newly generated `.md` file, I find the following line corresponding to my ggplot output  (which is sitting in a folder, right next to both the `.qmd`, and `.md` files:

`<img src="index.markdown_strict_files/figure-markdown_strict/unnamed-chunk-1-1.png" width="768" />`

For some reason, the website won't show up this image in the web page. 

Is that a way to fix this?

Thank you.

**Comment by cscheid** [comment]
What version of quarto are you using? `quarto --version`

**Comment by prvst** [comment]
version 1.2.313

I found a way around by replacing this:

`<img src="index.markdown_strict_files/figure-markdown_strict/unnamed-chunk-1-1.png" width="768" />`

with this

`![](index.markdown_strict_files/figure-markdown_strict/unnamed-chunk-1-1.png)`

I wonder if it's possible for the render engine to use this to represent image links

**Reply by cscheid** [reply]
Let me make sure I understand: your resulting `.md` has actual `<img src` tags in it?

I don't know how you're getting those; when attempting to render into markdown with the following example:

````
---
title: 'hello?'
format: markdown_strict
---

```{r}
library(ggplot2)
ggplot(mtcars, aes(mpg, cyl)) + geom_point()
```
````

I get the following output:

````
hello?
================

    library(ggplot2)
    ggplot(mtcars, aes(mpg, cyl)) + geom_point()

![](img-test.markdown_strict_files/figure-markdown_strict/unnamed-chunk-1-1.png)
````

**Comment by prvst** [comment]
Yes, that is correct.

I'm starting from a simple `.qmd` where I load a library, load the data, and then plot with ggplot2. I'm running `quarto render index.qmd` with no other parameters, and I get an `index.md` file. The `index.md` has the code snippets, and one `<img src` tag bellow. I tested multiple times, and the output is the same.

**Reply by prvst** [reply]
A little more information: I'm following [this instructions](https://quarto.org/docs/output-formats/hugo.html#workflow). And my `.qmd` file **has this parameter `format: hugo-md`**

**Reply by cscheid** [reply]
The `hugo-md` detail helps, thank you. Let me follow up internally and I'll get back to you.

**Reply by JackCaster** [reply]
Same issue here. Using Quarto `1.2.335`, Hugo `0.110`, and the `format: hugo-md` (or `format: hugo` ). The output `markdown_strict` does atttach the figures as markdown though. I am not sure what the difference is with `hugo-md`. Both seems to work fine with Hugo

EDIT: I don't see `markdown_strict` mentioned in the [docs](https://quarto.org/docs/output-formats/all-formats.html). Is this a supported format?

**Reply by cscheid** [reply]
Can you please share a full, minimal example in a repo so we can track down locally? Thanks.

**Comment by Yoshinobu-Ishizaki** [comment]
Same here. In my case, it is gfm.

I noticed that, 
- the very first image link is exclusively converted to `<img>` tag
- it does not matter whether it is just an image or plot

Here is an example.

````
---
title: 'quarto gfm image rendering test'
format: gfm
---

The very first image link is always converted to `<img>` tag by quarto.

![](test.png)

If you delete this line, the next image will be turned into `<img>`.

```{r}
library(ggplot2)
```

Though chunk output is suppressed, the first **markdown** image link is converted to `<img>` tag. 

```{r}
#| eval: false
ggplot(mtcars, aes(mpg, cyl)) + geom_point()
```

![](test_files/figure-commonmark/unnamed-chunk-2-1.png)


```{r}
# 2nd plot

ggplot(cars,aes(x = speed, y = dist)) + geom_point()
```

```{r}
version
```


```{r}
packageVersion("rmarkdown")
```

```{r}
system("quarto --version", intern = TRUE)
```
````

---

This was converted to,

````markdown
# quarto gfm image rendering test

The very first image link is always converted to `<img>` tag by quarto.

<img src="test.png" class="quarto-discovered-preview-image" />

If you delete this line, the next image will be turned into `<img>`.

``` r
library(ggplot2)
```

Though chunk output is suppressed, the first **markdown** image link is
converted to `<img>` tag.

``` r
ggplot(mtcars, aes(mpg, cyl)) + geom_point()
```

![](test_files/figure-commonmark/unnamed-chunk-2-1.png)

``` r
# 2nd plot

ggplot(cars,aes(x = speed, y = dist)) + geom_point()
```

![](test_files/figure-commonmark/unnamed-chunk-3-1.png)

``` r
version
```

                   _                                
    platform       x86_64-w64-mingw32               
    arch           x86_64                           
    os             mingw32                          
    crt            ucrt                             
    system         x86_64, mingw32                  
    status                                          
    major          4                                
    minor          2.3                              
    year           2023                             
    month          03                               
    day            15                               
    svn rev        83980                            
    language       R                                
    version.string R version 4.2.3 (2023-03-15 ucrt)
    nickname       Shortstop Beagle                 

``` r
packageVersion("rmarkdown")
```

    [1] '2.21'

``` r
system("quarto --version", intern = TRUE)
```

    [1] "1.3.290"
````

I could not figure out how to fix this.

**Reply by dragonstyle** [reply]
This is likely a result of a behavior we had earlier in the 1.3 version of the product (we would decorate the first image in the document with a class to make it easier for us to discover as the page preview). We changed this behavior a few weeks back and are no longer adding that class (I believe in 1.3.298) - can you try updating to the latest version and giving it a try?

**Reply by Yoshinobu-Ishizaki** [reply]
Using 1.3.333, it is fixed.
I'm sorry I did not try the most recent one. :bow:
Thank you.

**Reply by dragonstyle** [reply]
Great news! Thanks for reporting that initial problem and confirming the fix - we appreciate it!
